### ISSUE TITLE
preprocessing of texts and source code

### ISSUE THREAD

**Issue by liuhuigmail** [pcyin/tranX#24]
Great work on source code generation.
The details of the preprocessing of texts (naturel languges) and source code are missing from the paper. Would you kindly let me known what kind of preprossing has been conducted, e.g., unifying identifiers?  

Thanks.

Hui 
liuhui08@bit.edu.cn

**Comment by jason-hanling** [comment]
> Great work on source code generation.
> The details of the preprocessing of texts (naturel languges) and source code are missing from the paper. Would you kindly let me known what kind of preprossing has been conducted, e.g., unifying identifiers?
> 
> Thanks.
> 
> Hui
> [liuhui08@bit.edu.cn](mailto:liuhui08@bit.edu.cn)

have you solved the question ?
i am also curious about it

**Comment by liuhuigmail** [comment]
> > Great work on source code generation.
> > The details of the preprocessing of texts (naturel languges) and source code are missing from the paper. Would you kindly let me known what kind of preprossing has been conducted, e.g., unifying identifiers?
> > Thanks.
> > Hui
> > [liuhui08@bit.edu.cn](mailto:liuhui08@bit.edu.cn)
> 
> have you solved the question ?
> i am also curious about it

No. But I built a new dataset ([https://github.com/ds4an/CoDas4CG](https://github.com/ds4an/CoDas4CG)) and conduct a sequence of preprossing as I will :)

**Comment by jason-hanling** [comment]
> > > Great work on source code generation.
> > > The details of the preprocessing of texts (naturel languges) and source code are missing from the paper. Would you kindly let me known what kind of preprossing has been conducted, e.g., unifying identifiers?
> > > Thanks.
> > > Hui
> > > [liuhui08@bit.edu.cn](mailto:liuhui08@bit.edu.cn)
> > 
> > 
> > have you solved the question ?
> > i am also curious about it
> 
> No. But I built a new dataset (https://github.com/ds4an/CoDas4CG) and conduct a sequence of preprossing as I will :)

datasets/conala/dataset.py may do the preprocessing, i thought

**Comment by ShangwenWang** [comment]
@jason-hanling @liuhuigmail  

Hi, Professor Liu,

I am also interested about how to pre-process the data.
I note that the pre-processing is done by the dataset.py script (you are right).  I'd like to know what the files are like before pre-processing (conala-train.json). However, I found that the official webpage of CoNala (https://conala-corpus.github.io/) does not support downloading any more.
I wonder do you have something to share. Thanks!

**Comment by neubig** [comment]
I'm not sure why, but some people have been having trouble downloading the dataset on the chrome browser. Here's a direct link that should work. The dataset is still available:
http://www.phontron.com/download/conala-corpus-v1.1.zip

**Comment by ShangwenWang** [comment]
Oh great @neubig 
Thanks a lot.
